### DISCUSSION TITLE
How can I remove inputs in .map

### DISCUSSION THREAD

**Discussion by tktk0430** [react-hook-form/react-hook-form#3268] [Q&A] [answered]
I'm in trouble to handle removing inputs created by `<Redux store values>.map((item)=>...`

Here is example. Read local state **students** as redux store value.
```js
function App() {
  const [students, setStudents] = useState([
    { name: "tom" },
    { name: "bob" },
    { name: "alice" }
  ]);
  const { register } = useForm();

  return (
    <form>
      {students.map((item, idx) => {
        return (
          <div key={idx}>
            <label>First Name {idx}</label>
            <input
              type="text"
              name={`firstName[${idx}]`}
              ref={register}
              defaultValue={item.name}
            />
          </div>
        );
      })}
      <button onClick={() => setStudents([{ name: "bob" }, { name: "alice" }])}>
        imitate delete api (tom will be deleted)
      </button>
    </form>
  );
}
```
https://codesandbox.io/s/react-hook-form-validationschema-v6-forked-t5kzc?fontsize=14&hidenavigation=1&theme=dark
<img width="1116" alt="スクリーンショット 2020-10-26 20 47 26" src="https://user-images.githubusercontent.com/40733481/97168884-865de600-17cc-11eb-9e90-a4a214808e52.png">

First, there are inputs reflecting student names. Pushing button means posting delete api. That process like below.
1. At first, three students infos.
1. After delete api posted, Tom is deleted from store. To imitate that, I use `setStudents([{ name: "bob" }, { name: "alice" }])`
1. I expect there will be 2 forms showing 'bob' and 'alice'

But the result ends up bugging (plz try sandbox above). In real code (using real redux), it cause infinite loop.

In summary, I'd like to reflect redux store value, with reducing and also increasing amount of value.
How can i acheive this scenario ? 

Thanks.

**Comment by bluebill1049** [comment, marked as answer]
use `useFeildArray`?

**Reply by tktk0430** [reply]
Thank you for answer !

I know `useFieldArray` is very useful. But in this situation, I'm not sure which API I should use.
The forms need to detect state change, from `[{ name: "tom" }, { name: "bob" },  { name: "alice" }]` to `[{ name: "bob" },  { name: "alice" }]`.
So I think, I need something like `RESET` API to rerender forms reflecting new state by detecting state change.

Is there such a function in useFieldArray ?

**Reply by bluebill1049** [reply]
Do you want to try to get a unique Key for each input? see if that resolve your problem first.

**Reply by bluebill1049** [reply]
I give it a try now.

**Reply by bluebill1049** [reply]
try this: https://codesandbox.io/s/react-hook-form-validationschema-v6-forked-hg06m?file=/src/index.js

**Reply by tktk0430** [reply]
That's great !! Now I know that I need to get a unique Key for **`<input/>`**. Getting on `<div>` in `.map` is not enough. Thank you so much.

But still There is one problem on the sandbox above... New forms not reflect new states correctly, "tom" is still remaining.
I don't know if it is bug or not 😿

**Reply by tktk0430** [reply]
No, sorry, tom is removed.

**Reply by bluebill1049** [reply]
no worries : )

**Reply by tktk0430** [reply]
Thank you for supporting ! I love this lib so much !!!!

**Reply by bluebill1049** [reply]
No worries @tktk0430 Thank you so much for your support too! ❤️
